adapters/interlocked_win32.c

```diff
@@ -1,4 +1,318 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-#include "azure_c_util/interlocked.h"
\ No newline at end of file
+#include "azure_c_util/interlocked.h"
+#include "windows.h"
```

Comment:
We usually put windows before other things.

The order we try to stick to is:

- C standard headers (stdint.h, etc.)
- OS headers (windows, etc.)
- lib headers (azure_c_macro_utils, azure_c_utils, etc.)
Others